test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -92,8 +93,27 @@ public void Build_honors_UseStartup_with_string()
             using (host)
             {
                 host.Start();
-                var service = host.Services.GetServices<IApplicationLifetime>();
-                Assert.NotNull(service);
+                var services = host.Services.GetServices<IApplicationLifetime>();
+                Assert.NotNull(services);
+                Assert.NotEmpty(services);
+
+                await AssertResponseContains(server.RequestDelegate, "Exception from constructor");
+            }
+        }
+
+        [Fact]
+        public async Task ObjectPoolOfStringBuilderRegistered_EvenWhenStartupCtorThrows()
```

Comment:
`.Configure(app => { })` would be an adequate replacement for UseStartup in this test.